kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousTesterBuilder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test.tester;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import io.kroxylicious.proxy.config.Configuration;
+import io.kroxylicious.proxy.config.ConfigurationBuilder;
+import io.kroxylicious.test.ResponsePayload;
+import io.kroxylicious.test.server.MockServer;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+public class KroxyliciousTesterBuilder {
+
+    private String trustStoreLocation = null;
+    private String trustStorePassword = null;
+    private Function<Configuration, AutoCloseable> kroxyliciousFactory = DefaultKroxyliciousTester::spawnProxy;
+    private DefaultKroxyliciousTester.ClientFactory clientFactory = (clusterName, defaultConfiguration) -> new KroxyliciousClients(defaultConfiguration);
+    private ConfigurationBuilder configurationBuilder;
+
+    private Function<String, ConfigurationBuilder> mockConfigurationFunction;
+
+    private List<ResponsePayload> mockPayloads = new ArrayList<>();
+
+    public KroxyliciousTesterBuilder setKroxyliciousFactory(Function<Configuration, AutoCloseable> kroxyliciousFactory) {
+        this.kroxyliciousFactory = kroxyliciousFactory;
+        return this;
+    }
+
+    KroxyliciousTesterBuilder setClientFactory(DefaultKroxyliciousTester.ClientFactory clientFactory) {
+        this.clientFactory = clientFactory;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setConfigurationBuilder(ConfigurationBuilder configurationBuilder) {
+        this.configurationBuilder = configurationBuilder;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setTrustStoreLocation(String trustStoreLocation) {
+        this.trustStoreLocation = trustStoreLocation;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setTrustStorePassword(String trustStorePassword) {
+        this.trustStorePassword = trustStorePassword;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setMockConfigurationFunction(Function<String, ConfigurationBuilder> mockConfigurationFunction) {
+        this.mockConfigurationFunction = mockConfigurationFunction;
+        return this;
+    }
```

Comment:
I think this tells us there is something wrong with either how are setting up mocks or normal testers. 

Maybe this builder is the thing that should know about filters to add? However this isn't the PR to try and resolve the configuration of testers in.